x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/signals_alert_type.ts

```diff
@@ -96,7 +98,7 @@ export const signalsAlertType = ({ logger }: { logger: Logger }): SignalAlertTyp
             name,
             timeDetected: new Date().toISOString(),
             filter: esFilter,
-            maxDocs: maxSignals,
+            maxDocs: typeof maxSignals === 'number' ? maxSignals : 1000,
```

Comment:
Did you want this and not:

```ts
maxSignals != null ? maxSignals : 1000
```

Something seems a bit off though. Looking through a lot of this...

In the definition of: 
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/signals_alert_type.ts#L35

I see that it does have a default and this number is different than that one above.

```ts
maxSignals: schema.number({ defaultValue: 100 })
```

Since the alert params defines a default is it possible to still put it into a null/undefined state?

I see below this change:

```ts
maxSignals: number | undefined;
```

But I'm curious if we are seeing maxSignals capable of being set to null/undefined?

Unit tests have test cases covering the cases 
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas.test.ts#L347

Using the script `post_signal.sh` it posts the signal from `./signals/root_or_admin_1.json` which has this content and not a maxSignals defined which is a good e2e ad hoc test:

```ts
{
  "id": "rule-1",
  "description": "Detecting root and admin users",
  "index": ["auditbeat-*", "filebeat-*", "packetbeat-*", "winlogbeat-*"],
  "interval": "5m",
  "name": "Detect Root/Admin Users",
  "severity": "high",
  "type": "query",
  "from": "now-6m",
  "to": "now",
  "query": "user.name: root or user.name: admin",
  "language": "kuery",
  "references": ["http://www.example.com", "https://ww.example.com"]
}
```
Then on a read back: `./get_signal.sh rule-1`

I see it coming back ok:

```ts
{
  "id": "03c687ff-98dd-4c29-ab10-a6370cf86cf2",
  "name": "Detect Root/Admin Users",
  "alertTypeId": "siem.signals",
  "alertTypeParams": {
    "description": "Detecting root and admin users",
    "id": "rule-1",
    "index": [
      "auditbeat-*",
      "filebeat-*",
      "packetbeat-*",
      "winlogbeat-*"
    ],
    "from": "now-6m",
    "filter": null,
    "query": "user.name: root or user.name: admin",
    "language": "kuery",
    "savedId": null,
    "filters": null,
    "maxSignals": 100,
    "severity": "high",
    "to": "now",
    "type": "query",
    "references": [
      "http://www.example.com",
      "https://ww.example.com"
    ]
  },
  "interval": "5m",
  "enabled": true,
  "actions": [
    {
      "group": "default",
      "params": {
        "message": "SIEM Alert Fired",
        "level": "info"
      },
      "id": "23d5f855-e9f1-443a-91f5-2bf261f3d69c"
    }
  ],
  "throttle": null,
  "createdBy": "elastic",
  "updatedBy": "elastic",
  "apiKeyOwner": "elastic",
  "muteAll": false,
  "mutedInstanceIds": [],
  "scheduledTaskId": "eacd87d0-0567-11ea-8933-b7a2b6b3b4ee"
}
```

On an update everything looks ok as well which also does not set the maxSignal:

```ts
./update_signal.sh
./get_signal.sh rule-1
```

```ts
{
  "id": "03c687ff-98dd-4c29-ab10-a6370cf86cf2",
  "name": "A different name",
  "alertTypeId": "siem.signals",
  "alertTypeParams": {
    "description": "Changed Description of only detecting root user",
    "id": "rule-1",
    "index": [
      "auditbeat-*"
    ],
    "from": "now-6m",
    "filter": null,
    "query": "user.name: root",
    "language": "kuery",
    "savedId": null,
    "filters": null,
    "maxSignals": 100,
    "severity": "high",
    "to": "now-5m",
    "type": "query",
    "references": [
      "https://update1.example.com",
      "https://update2.example.com"
    ]
  },
  "interval": "50m",
  "enabled": false,
  "actions": [
    {
      "params": {
        "level": "info",
        "message": "SIEM Alert Fired"
      },
      "group": "default",
      "id": "23d5f855-e9f1-443a-91f5-2bf261f3d69c"
    }
  ],
  "throttle": null,
  "createdBy": "elastic",
  "updatedBy": "elastic",
  "apiKeyOwner": "elastic",
  "muteAll": false,
  "mutedInstanceIds": [],
  "scheduledTaskId": null
}
```

Then I created this payload in `/tmp/root_or_admin_1.json` where I tried to update it to null:

```ts
{
  "id": "rule-1",
  "description": "Detecting root and admin users",
  "index": ["auditbeat-*", "filebeat-*", "packetbeat-*", "winlogbeat-*"],
  "interval": "5m",
  "name": "Detect Root/Admin Users",
  "severity": "high",
  "type": "query",
  "from": "now-6m",
  "max_signals": null,
  "to": "now",
  "query": "user.name: root or user.name: admin",
  "language": "kuery",
  "references": ["http://www.example.com", "https://ww.example.com"]
}
```

And when running:

```ts
./update_signal.sh /tmp/root_or_admin_1.json
```

I get back this validation error:

```ts
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "child \"max_signals\" fails because [\"max_signals\" must be a number]",
  "validation": {
    "source": "payload",
    "keys": [
      "max_signals"
    ]
  }
}
```

Also with a `-1` it returns this:

```ts
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "child \"max_signals\" fails because [\"max_signals\" must be greater than 0]",
  "validation": {
    "source": "payload",
    "keys": [
      "max_signals"
    ]
  }
}
```

But a regular number in the payload such as: `100000` looks to do an update correctly:

```ts
{
  "id": "03c687ff-98dd-4c29-ab10-a6370cf86cf2",
  "name": "Detect Root/Admin Users",
  "alertTypeId": "siem.signals",
  "alertTypeParams": {
    "description": "Detecting root and admin users",
    "filter": null,
    "from": "now-6m",
    "query": "user.name: root or user.name: admin",
    "language": "kuery",
    "savedId": null,
    "filters": null,
    "index": [
      "auditbeat-*",
      "filebeat-*",
      "packetbeat-*",
      "winlogbeat-*"
    ],
    "maxSignals": 100000,
    "severity": "high",
    "to": "now",
    "type": "query",
    "references": [
      "http://www.example.com",
      "https://ww.example.com"
    ],
    "id": "rule-1"
  },
  "interval": "5m",
  "enabled": false,
  "actions": [
    {
      "params": {
        "level": "info",
        "message": "SIEM Alert Fired"
      },
      "group": "default",
      "id": "23d5f855-e9f1-443a-91f5-2bf261f3d69c"
    }
  ],
  "throttle": null,
  "createdBy": "elastic",
  "updatedBy": "elastic",
  "apiKeyOwner": "elastic",
  "muteAll": false,
  "mutedInstanceIds": [],
  "scheduledTaskId": null
}
```

Although the echo'ed return (my fault still) is using camelCase ... I still to need to transform eventually on the way back out. I'm just being expedient by relying on saved objects.

However, if you see this set somewhere to `undefined` or `null` let me know? I think if you have settled on your new number above of `1000` I would set that in the schema and you can remove the above code and anywhere where maxSignals is `null/undefined` as that shouldn't be possible as long as we keep our validation on the way in accepting defaults.